### PR TITLE
Balsamic report

### PR DESCRIPTION
**Pull request by keyvanelhami**
### This PR:

Added:
- CLI option to expand report generation for TGA and WES runs. Please see `balsamic report deliver --help`
- BALSAMIC now generates a custom HTML report for TGA and WES cases.

### CODE REVIEW

**Comment by keyvanelhami** [comment]
Please add following:

- [ ] A new a new table with following sample info: Analysdatum, Analyspanel, Provtyp. Also remove these headers from the "sammanfattning" table.
- [ ] Remove Provtyp header from the "Täckningsgrad för vald analyspanel" table
- [ ] Remove Balsamic version below the header
- [ ] Change the header to "Kvalitetsrapport"
- [ ] Add an text that this analysis is not accredited and maybe also add the Balsamic logo?

**Comment by hassanfa** [comment]
@keyvanelhami Is this correct: "BALSAMIC-analys är inte en ackrediterad metod." 😛

**Comment by hassanfa** [comment]
Or this: BALSAMIC är inte en SWEDAC ackrediterad metod.

**Comment by keyvanelhami** [comment]
Use this one =) 
"BALSAMIC-analys är inte en ackrediterad metod."

File: BALSAMIC/utils/qc_metrics.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    'required_metrics': [
+        'MEAN_TARGET_COVERAGE', 'MEDIAN_TARGET_COVERAGE',
+        'PCT_TARGET_BASES_50X', 'PCT_TARGET_BASES_100X',
+        'PCT_TARGET_BASES_200X', 'PCT_TARGET_BASES_500X',
+        'PCT_TARGET_BASES_1000X', 'FOLD_80_BASE_PENALTY'
+    ]
+}]
+
+
+def get_qc_metrics(case_name, output):
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:22]:
```suggestion
def get_qc_metrics(analysis_path):
```

File: BALSAMIC/utils/qc_metrics.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    Args:
+        case_name: given casename
+
+    Returns:
+        metrics_dict: dictionary
+
+    """
+    qc_data = {}
+    file_path = '/home/proj/production/cancer/cases/' + case_name + '/analysis/qc/multiqc_data/'
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:33]:
```suggestion

```

File: BALSAMIC/utils/qc_metrics.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            json_file = json.load(f)
+        for k in single_json['required_metrics']:
+            for j in json_file.keys():
+                sampleid = j.split('_')[1]
+                if sampleid not in qc_data:
+                    qc_data[sampleid] = {}
+                qc_data[sampleid][k] = json_file[j][k]
+
+    with open(output, 'w') as output_file:
+        json.dump(qc_data, output_file, indent=2)
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:48]:
```suggestion
return qc_data
```

File: BALSAMIC/utils/qc_metrics.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    Returns:
+        metrics_dict: dictionary
+
+    """
+    qc_data = {}
+    file_path = '/home/proj/production/cancer/cases/' + case_name + '/analysis/qc/multiqc_data/'
+
+    # Loop through json files
+    for single_json in multiple_json:
+        file_name = file_path + single_json['file_name']
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:37]:
```suggestion
        file_name = os.path.join(analysis_path, "qc", "multiqc_data", single_json['file_name'])
```

File: BALSAMIC/utils/qc_metrics.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                if sampleid not in qc_data:
+                    qc_data[sampleid] = {}
+                qc_data[sampleid][k] = json_file[j][k]
+
+    with open(output, 'w') as output_file:
+        json.dump(qc_data, output_file, indent=2)
+
+
+if __name__ == '__main__':
+    get_qc_metrics(*sys.argv[1:])
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:52]: No need for this. We will import get_qc_metrics and provide it with pass
```suggestion

```

File: BALSAMIC/utils/qc_metrics.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        'PCT_TARGET_BASES_1000X', 'FOLD_80_BASE_PENALTY'
+    ]
+}]
+
+
+def get_qc_metrics(case_name, output):
+    """Reads picard metrics for a particular case and returns a new dict
+
+    Args:
+        case_name: given casename
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:26]:
```suggestion
        analysis_path: analysis result path, e.g. /path/case_name/analysis
```

File: BALSAMIC/utils/qc_metrics.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    Returns:
+        metrics_dict: dictionary
```
Comment (hassanfa) [inline BALSAMIC/utils/qc_metrics.py:29]:
```suggestion
        qc_data: a dictionary of extracted metrics
```

**Review by hassanfa — COMMENTED** [review]
Nice work! What do you think about the following changes?

File: BALSAMIC/utils/qc_metrics.py (hunk 1 above)
Comment (ashwini06) [reply BALSAMIC/utils/qc_metrics.py]: Fixed

File: BALSAMIC/utils/qc_metrics.py (hunk 2 above)
Comment (ashwini06) [reply BALSAMIC/utils/qc_metrics.py]: Fixed

File: BALSAMIC/utils/qc_metrics.py (hunk 6 above)
Comment (ashwini06) [reply BALSAMIC/utils/qc_metrics.py]: Fixed

File: BALSAMIC/utils/qc_metrics.py (hunk 5 above)
Comment (ashwini06) [reply BALSAMIC/utils/qc_metrics.py]: Fixed

**Comment by ashwini06** [comment]
> Nice work! What do you think about the following changes?

Thanks @hassanfa  for your suggestions. Fixed and added tests In latest commit.

File: BALSAMIC/assets/report_template/balsamic_report.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+Analyspanel
+: {{ meta.gene_panel_name }} 
+
+Analyskod
+: {{ meta.apptag }} 
+
+Analysdatum
+: {{ meta.config_date }}
+
+Bioinformatiskanalys:
```
Comment (ashwini06) [inline BALSAMIC/assets/report_template/balsamic_report.md:15]:
```suggestion
Bioinformatikanalys:
```

File: BALSAMIC/utils/qc_metrics.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import json
+import sys
```
Comment (ashwini06) [inline BALSAMIC/utils/qc_metrics.py:2]:
```suggestion

```

**Review by ashwini06 — COMMENTED** [review]
very minor comments and looks good  🚀  👍

File: BALSAMIC/assets/report_template/balsamic_report.md (hunk 8 above)
Comment (hassanfa) [reply BALSAMIC/assets/report_template/balsamic_report.md:15]: I think we are both wrong :-p  It should be two words. "Bioinformatisk analys" 😁 I asked our Swedish colleagues. :-)

**Review by ashwini06 — APPROVED** [review]
👍

**Review by hassanfa — APPROVED** [review]
Approving Keyvan and Ashwini's part.

**Pull request merged**